File: push/src/main/java/com/nihility/service/NotificationsRevivalAbility.java
package com.nihility.service;

import android.os.Build;

import androidx.annotation.RequiresApi;

import com.xiaomi.push.revival.NotificationsRevivalForSelfUpdated;

@RequiresApi(api = Build.VERSION_CODES.M)
public class NotificationsRevivalAbility implements XMPushServiceListener {
    private final NotificationsRevivalForSelfUpdated notificationsRevival;

    public NotificationsRevivalAbility(NotificationsRevivalForSelfUpdated notificationsRevival) {
        this.notificationsRevival = notificationsRevival;
    }

    @Override
    public void created() {
        notificationsRevival.initialize();
    }

    @Override
    public void destroy() {
        notificationsRevival.close();
    }
}

File: xmsf/shell/src/main/java/io/github/magisk317/mipush/service/NotificationsRevivalAbility.kt
package io.github.magisk317.mipush.service

import android.os.Build
import androidx.annotation.RequiresApi
import com.xiaomi.push.revival.NotificationsRevivalForSelfUpdated

@RequiresApi(api = Build.VERSION_CODES.M)
class NotificationsRevivalAbility(
    private val notificationsRevival: NotificationsRevivalForSelfUpdated
) : XMPushServiceListener {
    override fun created() {
        notificationsRevival.initialize()
    }

    override fun destroy() {
        notificationsRevival.close()
    }
}
